x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformStats.java

```diff
@@ -92,40 +104,38 @@ public TransformStats(
         @Nullable String reason,
         @Nullable NodeAttributes node,
         TransformIndexerStats stats,
-        TransformCheckpointingInfo checkpointingInfo
+        TransformCheckpointingInfo checkpointingInfo,
+        TransformHealth health
     ) {
         this.id = Objects.requireNonNull(id);
         this.state = Objects.requireNonNull(state);
         this.reason = reason;
         this.node = node;
         this.indexerStats = Objects.requireNonNull(stats);
         this.checkpointingInfo = Objects.requireNonNull(checkpointingInfo);
+        this.health = Objects.requireNonNull(health);
```

Comment:
`requireNonNull` here is inconsistent with `optionalConstructorArg` in the parser definition.